src/replication.c

```diff
@@ -2308,6 +2474,441 @@ char *sendCommandArgv(connection *conn, int argc, char **argv, size_t *argv_lens
     return NULL;
 }
 
+/* Replication: Replica side.
+ * Returns an sds represent this replica port to be used by the primary (mostly
+ * for logs) */
+sds getReplicaPortString(void) {
+    long long replica_port;
+    if (server.replica_announce_port) {
+        replica_port = server.replica_announce_port;
+    } else if (server.tls_replication && server.tls_port) {
+        replica_port = server.tls_port;
+    } else {
+        replica_port = server.port;
+    }
+    return sdsfromlonglong(replica_port);
+}
+
+/* Replication: Replica side.
+ * Free replica's local replication buffer */
+void freePendingReplDataBuf(void) {
+    if (server.pending_repl_data.blocks) {
+        listRelease(server.pending_repl_data.blocks);
+    }
+    server.pending_repl_data.blocks = NULL;
+    server.pending_repl_data.len = 0;
+}
+
+/* Replication: Replica side.
+ * Upon rdb-sync failure, close rdb-connection, reset repl-state, reset
+ * provisional primary struct, and free local replication buffer. */
+void abortRdbConnectionSync(void) {
+    serverAssert(server.repl_rdb_conn_state != REPL_RDB_CONN_STATE_NONE);
+    serverLog(LL_WARNING, "Aborting dual connection sync");
```

Comment:
I think `LL_NOTICE` is better.
```suggestion
    serverLog(LL_NOTICE, "Aborting dual connection sync");
```

Reply:
LL_WARNING is intended to warn operators about material issues they need to take action on, I think this can happen for other issues. So LL_NOTICE makes sense.

Reply:
👍